python/lsst/pipe/base/graph/graph.py

```diff
@@ -1656,3 +1658,88 @@ def init_output_run(self, butler: LimitedButler, existing: bool = True) -> None:
         self.write_configs(butler, compare_existing=existing)
         self.write_packages(butler, compare_existing=existing)
         self.write_init_outputs(butler, skip_existing=existing)
+
+    def get_refs(
```

Comment:
@andy-slac what do you think of this? As for intermediates, I've lost track of how I find them.

Reply:
Looks good to me, if that works for you. Intermediates should be intersection of inputs and outputs?